File: repo/frontend/src/types/schedule.ts
export interface Schedule {
  id: number
  userId: number
  title: string
  description: string
  startTime: string
  endTime: string
  dayOfWeek: number
  isRecurring: boolean
  color: string
  contentRecordId: number
  quizPaperId: number
}

export interface LockedPeriod {
  id: number
  title: string
  startTime: string
  endTime: string
  reason: string
  dayOfWeek?: number
}

export interface ScheduleChangeJournal {
  id: number
  scheduleId: number
  changeType: string
  previousState: string
  newState: string
  isUndone: boolean
  sequenceNumber: number
  createdAt: string
}

File: repo/backend/src/main/java/com/scholarops/model/entity/Schedule.java
package com.scholarops.model.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "schedules")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Schedule {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JsonIgnore
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "start_time", nullable = false)
    private LocalDateTime startTime;

    @Column(name = "end_time", nullable = false)
    private LocalDateTime endTime;

    @Column(name = "day_of_week")
    private Integer dayOfWeek;

    @Column(name = "is_recurring", nullable = false)
    @Builder.Default
    private Boolean isRecurring = false;

    @Column(length = 7)
    private String color;

    @Column(name = "content_record_id")
    private Long contentRecordId;

    @Column(name = "quiz_paper_id")
    private Long quizPaperId;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
